applications/StructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/error_mesh_criteria.h

```diff
@@ -0,0 +1,327 @@
+    {
+        /**
+         * error_mesh_tolerance: The tolerance in the convergence criteria of the error
+         * error_mesh_constant: The constant considered in the remeshing process
+         * minimal_size: The minimal size of element for the mesh
+         * maximal_size: The maximum size of element for the mesh
+         * error: The target error
+         * penalty_normal: The penalty used in the normal direction (for the contact patch)
+         * penalty_tangential: The penalty used in the tangent direction (for the contact patch)
+         * echo_level: The verbosity
+         * set_number_of_elements: If the number of elements will be forced or not
+         * number_of_elements: The estimated/desired number of elements
+         * average_nodal_h: If the nodal size to consider will be averaged over the mesh
+         */
+        Parameters default_parameters = Parameters(R"(
+        {
+            "error_mesh_tolerance" : 5.0e-3,
+            "error_mesh_constant"  : 5.0e-3,
+            "error_strategy_parameters":
+            {
+                "minimal_size"                        : 0.01,
+                "maximal_size"                        : 1.0,
+                "error"                               : 0.01,
+                "penalty_normal"                      : 1.0e4,
+                "penalty_tangential"                  : 1.0e4,
+                "echo_level"                          : 0,
+                "set_number_of_elements"              : false,
+                "number_of_elements"                  : 1000,
+                "average_nodal_h"                     : false
+            }
+        })" );
+        
+        mThisParameters.ValidateAndAssignDefaults(default_parameters);
+        
+        mErrorTolerance = mThisParameters["error_mesh_tolerance"].GetDouble();
+        mConstantError = mThisParameters["error_mesh_constant"].GetDouble();
+        
+    }
+
+    ///Copy constructor 
+    ErrorMeshCriteria( ErrorMeshCriteria const& rOther )
+      :BaseType(rOther)
+      ,mErrorTolerance(rOther.mErrorTolerance)
+      ,mConstantError(rOther.mConstantError)
+    {
+    }
+
+    /// Destructor
+    ~ErrorMeshCriteria() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * @brief This function initialize the convergence criteria
+     * @param rModelPart The model part of interest
+     */
+    void Initialize(ModelPart& rModelPart) override
+    {
+        BaseType::Initialize(rModelPart);
+
+        const bool meshing_application_compiled = KratosComponents<Variable<Vector>>::Has("MMG_METRIC");
+        KRATOS_WARNING_IF("CalculateMetric", !meshing_application_compiled) << "Please compile the MeshingApplication in order to compute the metric" << std::endl;
+
+        // Initializing metric (if possible)
+        if (meshing_application_compiled) {
+            const Variable<Vector>& metric_variable = KratosComponents<Variable<Vector>>::Get("MMG_METRIC");
```

Comment:
what happens if meshing is not compiled?
=> does this criteria still work?

I am thinking you could compile this conditional, if the necessary requirements (Meshing and MMG from what I understood) are there => this can easily be checked in cmake
Then you could throw an error in the conv-criteria-factory if StructMech doesn't have this criteria

Or thinking even further, why not putting it in the MeshingApp? (Sorry if it was there before, I don't remember)

Reply:
Well, I suppose I should update the criteria and just check the error, the thing is that this PR depends of several PR's (because @RiccardoRossi  make me separate them), and until I fixed the others I cannot correct this one